testbed/testbed/mock_backend.go

```diff
@@ -161,6 +177,21 @@ func (tc *MockTraceConsumer) Capabilities() consumer.Capabilities {
 }
 
 func (tc *MockTraceConsumer) ConsumeTraces(_ context.Context, td ptrace.Traces) error {
+	switch tc.backend.decision {
+	case 1:
+		if err := randomNonPermanentError(); err != nil {
+			return err
+		}
+	case 2:
+		if err := randomPermanentError(); err != nil {
+			if tc.backend.isRecording {
+				tc.backend.DroppedTraces = append(tc.backend.DroppedTraces, td)
+			}
+			return err
+		}
+	case 3:
+		// TODO
```

Comment:
What does the `// TODO` here refer to doing?

Reply:
Yeah, so I'm planning to add another scenario.
I would put probably put something like `time.Sleep`, drop the data and return the `server is busy` error. This was we will test if `sendin_queue` and `retry_on_failure` are functioning properly if the destination is busy.
I will add this as a comment.